api/proto/teleport/legacy/types/types.proto

```diff
@@ -5477,6 +5477,42 @@ message WindowsDesktopSpecV3 {
   Resolution ScreenSize = 5 [(gogoproto.jsontag) = "screen_size,omitempty"];
 }
 
+// DynamicWindowsDesktopV1 represents a dynamic windows host for desktop access.
+message DynamicWindowsDesktopV1 {
```

Comment:
We don't need it in List Resources and such because they are only used to create WindowsDesktops that will show up in those APIs already. We do want to add it to terraform provider and operator, though.

Reply:
For the operator there's [open issue](https://github.com/kubernetes/kubernetes/issues/124154) that prevents us from using 153-style resources there. That's one of the reasons to put this into legacy package.

Reply:
We'd like to support Terraform and the Kube operator, but if I had to pick one then Terraform is probably more important.

I think it makes sense to keep dynamic registration as similar as we can across all resources. If Dynamic registration for apps and databases doesn't use 153-style resources then we might as well be consistent.

Reply:
I don't _think_ that we will heartbeat on them TBH, they are meant to be created/managed by user (or terraform but still) not by our service so they probably don't need heartbeat unless I miss something